src/tile/tile.ts

```diff
@@ -140,3 +140,98 @@ export const findTileByCoordinate = (
   board: Board,
   coords: TileCoordinate
 ): PlayableTile => board[coords.x][coords.y] as PlayableTile;
+
+export const getTileName = (tile: PlayableTile): string =>
+  `${tile.symbol}-${tile.color}`;
+
+export const findLastPlayableTile = (
+  line: Tile[]
+): PlayableTile | undefined => {
+  const playable = line.findLast((tile) => tile != undefined);
+  return playable as PlayableTile | undefined;
+};
+
+export const findFirstPlayableTile = (line: Tile[]): Tile | undefined => {
+  const playable = line.find((tile) => tile != undefined);
+  return playable as PlayableTile | undefined;
+};
+
+export const checkIfCoordsExist = (board: Board, coords: TileCoordinate) =>
+  board[coords.x] &&
+  board[coords.x][coords.y] &&
+  board[coords.x][coords.y].symbol
+    ? true
+    : false;
+
+interface Siblings {
+  topLeft: TileCoordinate | null;
+  topRight: TileCoordinate | null;
+  next: TileCoordinate | null;
+  bottomRight: TileCoordinate | null;
+  bottomLeft: TileCoordinate | null;
+  previous: TileCoordinate | null;
+}
+
+export const findSiblings = (
+  board: Board,
+  tileCoords: TileCoordinate,
+  player?: Player
+): Siblings => {
+  const siblingsCoords = {
+    topLeft: {
+      x: tileCoords.x - 1,
+      y: tileCoords.y,
+    },
+    topRight: {
+      x: tileCoords.x - 1,
+      y: tileCoords.y + 1,
+    },
+    next: {
+      x: tileCoords.x,
+      y: tileCoords.y + 1,
+    },
+    bottomRight: {
+      x: tileCoords.x + 1,
+      y: tileCoords.y + 1,
+    },
+    bottomLeft: {
+      x: tileCoords.x + 1,
+      y: tileCoords.y,
+    },
```

Comment:
I believe it should be:

```suggestion
    bottomRight: {
      x: tileCoords.x + 1,
      y: tileCoords.y,
    },
    bottomLeft: {
      x: tileCoords.x + 1,
      y: tileCoords.y - 1,
    },
```

Reply:
I realize it's actually more complex than that.
Since `null` tiles are not the same width as non-`null` tiles, you are distorting the hexagonal space.
TDLR the bottom-half of the board does not have the same connexity as the top-half.

Reply:
since the demo is in 2 hours, we agreed to merge the PR as is, knowing that this code is partially broken